overlord/snapstate/snapstate.go

```diff
@@ -2565,6 +2578,40 @@ func RemoveMany(st *state.State, names []string) ([]string, []*state.TaskSet, er
 	return removed, tasksets, nil
 }
 
+func findNonExistentSnaps(ctx context.Context, st *state.State, names []string, userID int) ([]string, error) {
+	user, err := userFromUserID(st, userID)
+	if err != nil {
+		return nil, err
+	}
+
+	storeServ := Store(st, nil)
```

Comment:
I don't think this should be using the store to check if the snap exists, it should check if the snap exists locally as an installed snap, this may mean we want to spruce up the error message to make it clear that the snap just isn't installed and not make mention of whether the snap exists in the store or is installable, etc.

Reply:
*it should _just_ check

Reply:
I agree, moreover this is partially handled already below when we call removeTasks(..), and ignore snap.NotInstalledError, so seems that it just needs special casing below.

Reply:
My understanding was that we wanted to preserve the behaviour when doing `snap remove installed-snap already-rm-snap` so that it wouldn't fail and only `snap remove installed-snap non-existent-snap` should fail. For that we have to distinguish between non-existent and uninstalled snaps in some way. Michael recently mentioned a snap database in the advisor pkg which could replace the store access and seems like a good fit for this. It's updated every day which seems enough, in the worst case we fail a `snap remove` if the user includes a very new uninstalled snap . What do you guys think?

Reply:
Ah, sorry, I misunderstood what we wanted and didn't see the bug report. Looking at the book report it seems to me that maybe all we need is to validate snap names (we already have a helper for this)?

Reply:
My bad, I forgot to put link in the description. I I guess that would prevent most mistakes unless the user omits both the dashes and equals like `snap remove validSnap revision blah` which is a bit more far fetched. I'll make the change then, thanks